Cargo.toml

```diff
@@ -6,14 +6,18 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-argh = { version = "0.1.4", default-features = false }
-reqwest = { version = "0.11.3", default-features = false, features = ["blocking", "default-tls"] }
-serde = { version = "1.0.125", default-features = false, features = ["derive"] }
```

Comment:
I don't bump version numbers in Cargo.toml unless I need something that has only been added in that version. I'm a big believer in minimal versions, even for binary crates. The lockfile covers the common cases.

Reply:
The rationale for doing it with binaries is that a common failure mode for users installing tools is during install without `--locked` where some upstream dependency breaks semver. While this is user error, I find for binaries specify x.y.z, letting dependabot handle keeping that up to date and making sure the updates are tested is a healthy release approach. Put another way, feel your user's pain before they do when you can. You're probably much more knowledgeable than me on this, so I'm curious how that squares with your experience.